app/views/spree/admin/orders/_table.html.haml

```diff
@@ -14,7 +14,7 @@
       %th
         = t(:products_distributor)
 
-      - columns = ['completed_at', 'number', 'state', 'payment_state', 'shipment_state', 'email', 'bill_address_lastname', 'total']
+      - columns = ['completed_at', 'number', 'state', 'payment_state', 'shipment_state', 'email', 'billing_address_name', 'total']
```

Comment:
This `billing_address_name` uses a [Ransack scope](https://activerecord-hackery.github.io/ransack/going-further/other-notes/#using-scopesclass-methods). 

Note, it can't be called `bill_address_name` because then Ransack doesn't recognise the scope and treats it as an ordinary column. The query then fails because it tries to search by `spree_addresses.name` which isn't a real column.

Another solution might be to not use a Ransack scope and pass two sort columns via the `q[s]` paramter to Ransack, something like...

```
 columns = [
  'completed_at',
  'number',
  'state',
  'payment_state',
  'shipment_state',
  'email',
  ['name', ['bill_address.lastname', 'bill_address.firstname']],
  'total'
]
```